src/averageAges.js

```diff
@@ -55,10 +75,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filteredPeople = people.filter(person => {
+    const hasMother = people.find(mother => mother.name === person.mother);
+
+    return hasMother && (onlyWithSon ? person.sex === MALE : true);
+  });
+
+  const childrenTotalAge = filteredPeople
+    .reduce((acc, { born, mother: motherName }) => {
+      const motherBirthYear = people
+        .find(({ name }) => name === motherName).born;
```

Comment:
you put methods on the new line if the're few of them, in your case you can just put condition on the new line and wrap it with braces, should be more readable